laravel-app/app/Services/PostService.php

```diff
@@ -63,4 +64,35 @@ public function sharePost(UserPost $post): UserPost
 
         return $sharedPost;
     }
+
+    /**
+     * To update the image on post
+     */
+    public function updatePostImage(UserPost $post, UploadedFile $image): bool
+    {
+        try {
+            if ($image != null) {
+                $imagePath = $image->store('images', 'public');
+
+                if ($post->user !== null) {
+                    $userId = $post->user->id;
+                } else {
+                    $userId = null; // Handle the case where $post->user is null
+                }
+
+                $post->media()->updateOrCreate(
+                    [
+                        'user_id' => $userId,
+                    ],
+                    [
+                        'file_path' => $imagePath,
+                    ],
+                );
+            }
+
+            return true;
+        } catch (\Exception $e) {
+            return false;
+        }
+    }
```

Comment:
Prevent using else in return statement.

https://medium.com/@andrerferrer/unnecessary-else-after-return-no-else-return-43e697575dc1#:~:text=So%2C%20what%20does%20it%20mean,NEED%20the%20%60else%60%20statement.&text=This%20is%20a%20case%20in,if%20the%20number%20is%20even.

Reply:
yes sir 🙇